### DISCUSSION TITLE
Get JWT string token without encryption

### DISCUSSION THREAD

**Discussion by georgekrax** [nextauthjs/next-auth#3424] [Help]
Hello all!

I am new to `next-auth` and I am trying to access the JWT token without being previously encrypted? The `state` cookie, has an encrypted version of it. Any idea? Also, is there a Discord server for Q&A?

Thanks for your help!

**Comment by balazsorban44** [comment]
We don't have a Discord server for making answers publicly searchable by anyone :slightly_smiling_face:.

Do you have a code example of what you are trying to do?

**Reply by georgekrax** [reply]
No, I just couldn't find a way to decode the content of the the `session` cookie through the `jwt.io` debugger. So, I guessed that the token there is encrypted, and turned into a `JWE`. Is there a way to get the decrypted JWT from this token, in order to debug it online? I am just curious, I do not just want to access the content of the token, that's why.

**Reply by balazsorban44** [reply]
Yes, the session is encrypted. You can use the `import { getToken } from "next-auth/jwt` method to read it's value server-side.

You get the same result of what you are returning from your `jwt` callback.

**Reply by georgekrax** [reply]
However, I would to read its decoded value in my own seperate backend server? Should I install `next-auth` in my backend project and call `getToken()` in there? Or, should I implement custom `encode()`, `decode()` functions in the `jwt` and use the custom `decode()` one in my backend, without having to install the package two times?
 
Also, why do you use `jose` and not `jsonwebtoken`? How can a create and encrypt a `JWE` with `jsonwebtoken`, like you and with `jose`?

Thanks in advance!

**Reply by balazsorban44** [reply]
You shouldn't use the next-auth JWT outside of the app. use your provider's JWT instead. How you would decode that, depends on your provider of choice.

**Reply by georgekrax** [reply]
> Also, why do you use `jose` and not `jsonwebtoken`? How can a create and encrypt a `JWE` with `jsonwebtoken`, like you and with `jose`?

Oh, okay. Thank you for your advice. Can you answer to those questions too?

**Reply by balazsorban44** [reply]
Because we wanted to..? 😅

I recommend looking up their documentation.

**Reply by georgekrax** [reply]
Yeah, of course. I mean like are there any specific advantages and features that `jose` has instead of `jsonwebtoken`?`

**Reply by balazsorban44** [reply]
When I started working on the library, `jose` was already a dependency. It is well-maintained with great docs, and it's author really makes sure it remains high quality.

You can use whatever suits you best in your app.

**Reply by georgekrax** [reply]
Do you know if I can use `jsonwebtoken` to encrypt the generated JWT? This functionality is available to `jose`, but is there a way to implement it with `jsonwebtoken`?
